### DISCUSSION TITLE
REST API responses for workflow runs (`/actions/runs?created=2024-02-01`) seem to cut off after 1000 responses always

### DISCUSSION THREAD

**Discussion by GovindarajanNagarajan-TomTom** [octokit/.github#81] [Q&A] [answered]
We have a bunch of repos that run a lot of workflows per day and we want to track the job durations and the workflow durations of those Repositories. 
To wit, we use the GH REST API to obtain this information |
```
https://api.github.com/repos/org-name/repo-name/actions/runs\?created\=2024-03-08 -H"Accept: application/vnd.github+json" -H "X-Github-Api-Version: 2022-11-28" -H "Authorization: Bearer github_personal_token"  -v
```
In the response, I get the total count for that day as `1141` , but as I iterate through the pages, the response never has more than 1000 entries. With page=11 and per_page set to 100, the reponse for the 11th page is 
```
{
  "total_count": 0,
  "workflow_runs": [

  ]
}

```
and the `link` header is missing in the response.

Is there a well defined cutoff defined in GH REST API documentation ? Or is this a problem with personal tokens ? Or is this a problem with rate-limiting ? 
Would using an Github App installation fix the issue ?

**Comment by gr2m** [comment]
This seems to be an API question, not an @octokit question.

Can you reproduce the problem requesting the 11th page using curl or `gh api`? It would help support to replicate the problem and respond.

I know that search responses are limited to 1000 results. If https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository is also limited to 1000 results, we should make sure the docs get updated.

**Comment by GovindarajanNagarajan-TomTom** [comment]
I was able to replicate this with the GH REST API 
```
$ curl https://api.github.com/repositories/redacted_repo_id_here/actions/runs\?created\=2024-03-01\&per_page=100\&page=11 -H"Accept: application/vnd.github+json" -H "X-Github-Api-Version: 2022-11-28" -H "Authorization: Bearer redacted_personal_token" -S -D headerout
{
  "total_count": 0,
  "workflow_runs": [

  ]
}
```

The total count reported for this API in the first 1-10 pages was `1158`. So it looks like the REST API clips all responses after the 1000th entry.
If you have more than a 1000 runs for a single job in the repo, page 11 on seems to be giving `[]` outputs

**Reply by gr2m** [reply]
The Actions team is looking into it. No promises. First insight is that it likely depends on the `created` query parameter being set. I'll let you know once the actions team gets back to me.

**Reply by GovindarajanNagarajan-TomTom** [reply]
Thank you :) . I just want to understand what the behaviour is and if something is missing in the docs, to be added as I wasn't sure why some tooling we developed wasn't working as intended and we had gaps in our data .

**Reply by gr2m** [reply, marked as answer]
The documentation has been updated

> This API will return up to 1,000 results for each search when using the following parameters: actor, branch, check_suite_id, created, event, head_sha, status.

Sorry, but there is indeed a hard cap at 1,000 results.

**Reply by GovindarajanNagarajan-TomTom** [reply]
Thank you, atleast the behaviour is documented now, so we can look into alternatives :)
